pkg/cloud/amazon/aws_kms.go

```diff
@@ -185,7 +185,6 @@ func MakeAWSKMS(ctx context.Context, uri string, env cloud.KMSEnv) (cloud.KMS, e
 			return nil, errors.New(
 				"implicit credentials disallowed for s3 due to --external-io-disable-implicit-credentials flag")
 		}
-		addLoadOption(config.WithSharedConfigProfile(config.DefaultSharedConfigProfile))
```

Comment:
nice sleuthing! is it possible to add a unit test that would have failed due to this bug? See the Cloud Unit test nightly  suite: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_Nightlies_CloudUnitTests&branch_Cockroach_Nightlies=129938&tab=buildTypeStatusDiv